### PR TITLE
ISPN-5095 EntryIterator should only iterate on data container when local

### PR DESCRIPTION
**Pull request by wburns**
and no loader used

https://issues.jboss.org/browse/ISPN-5095

### CODE REVIEW

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity Infinispan :: Pull requests monitor [Build 8277/wburns:ISPN-5095](http://ci.infinispan.org/viewLog.html?buildId=22756&buildTypeId=bt9&guest=1) for 8812208d3918cffb228afda78edd7aa45825d235
Tests: 11173, 1 failed (0 new), 22 ignored. Build time: 02:23:02

Failed tests detected
* (since 22-Dec-2014) org.infinispan.test.integration.as.InfinispanLuceneDirectoryIT



Error message is logged

Build duration (secs) is too large: 8624

File: core/src/main/java/org/infinispan/iteration/impl/LocalEntryRetriever.java [hunk 1]
```diff
@@ -246,6 +247,104 @@ protected boolean shouldUseLoader(Set<Flag> flags) {
...
+               cache.remove(prev.getKey());
+            }
+
+            @Override
+            public void close() {
+               // Nothing to do here
+            }
+            
+         };
+      }
```
Comment (tristantarrant) [inline core/src/main/java/org/infinispan/iteration/impl/LocalEntryRetriever.java]: retrieveEntries() is already huge, can't this code be refactored in submethods please ?

Comment (wburns) [inline core/src/main/java/org/infinispan/iteration/impl/LocalEntryRetriever.java]: Sure I can change it up.

**Comment by wburns** [comment]
Moved out the inner class and also update the commit message as I had transposed to the 0 and the 9.

**Comment by tristantarrant** [comment]
Pushed, thanks

**Pull request closed without merging**